apps/backend/src/opencage/opencage.controller.ts

```diff
@@ -0,0 +1,31 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import * as https from 'https';
+
+@Controller('api/v1/location')
+export class OpencageController {
+  @Get()
+  async reverseGeocoding(@Query('latitude') latitude: string, @Query('longitude') longitude: string): Promise<any> {
+    if (!latitude || !longitude) {
+      throw new Error('Latitude and longitude must be provided.');
+    }
+
+    const url = `https://api.opencagedata.com/geocode/v1/json?q=${latitude}+${longitude}&key=${process.env.OPENCAGE_KEY}`;
+
+    return new Promise((resolve, reject) => {
+      https.get(url, (response) => {
+        let data = '';
+
+        response.on('data', (chunk) => {
+          data += chunk;
+        });
+
+        response.on('end', () => {
+
+          resolve(JSON.parse(data)?.results.map(address => ({region: address.components.city, subregion: address.components.state})));
```

Comment:
i live in Portugal and i ended up getting this ![Screenshot 2024-03-29 at 11 18 19](https://github.com/animavita/animavita/assets/19699724/a6be6caf-7bc1-4d57-af4f-6e2995e3a13f)

this works perfectly with Brazilian addresses and our app is first supposed to work only in Brazil. however, we do have plans to expand it abroad so it would be great to support other formats. I'm not sure about all formats, but I believe for now we could fall back the city to `municipality` and the state to `county`


![Screenshot 2024-03-29 at 11 24 31](https://github.com/animavita/animavita/assets/19699724/f16b3b05-bd49-4245-afb3-ca4250da5ced)

Reply:
I tried to create a GeolocationService interface and use the following snippet in the GeolocationModule providers:


``` typescript
providers: [
    { provide: GeolocationService, useClass: OpenCageService },
],
```
However, this did not work and resulted in the following error:

```
'GeolocationService' only refers to a type, but is being used as a value here.
```
Although I was unable to resolve this error, I was still able to refactor the code to improve it.

Reply:
Yeah but it looks like there's no city and state properties available in the return payload for European addresses